src/podcast_processor/transcribe.py

```diff
@@ -158,3 +164,143 @@ def get_segments_for_chunk(self, chunk_path: str) -> List[TranscriptionSegment]:
             self.logger.debug(f"Got {len(segments)} segments")
 
             return segments
+
+
+class GroqTranscriptionSegment(BaseModel):
+    start: float
+    end: float
+    text: str
+
+
+class GroqTranscriptionResponse(BaseModel):
+    segments: List[GroqTranscriptionSegment]
+
+
+class GroqWhisperTranscriber(Transcriber):
+
+    def __init__(self, logger: logging.Logger, config: GroqWhisperConfig):
+        self.logger = logger
+        self.config = config
+        self.api_url = "https://api.groq.com/openai/v1/audio/transcriptions"
+        self.max_retries = config.max_retries
+        self.initial_backoff = config.initial_backoff
+        self.backoff_factor = config.backoff_factor
+
+    def transcribe(self, audio_file_path: str) -> List[Segment]:
+        self.logger.info("Using Groq whisper")
+        audio_chunk_path = audio_file_path + "_parts"
+
+        chunks = split_audio(
+            Path(audio_file_path), Path(audio_chunk_path), 12 * 1024 * 1024
+        )
+
+        all_segments: List[GroqTranscriptionSegment] = []
+
+        for chunk in chunks:
+            chunk_path, offset = chunk
+            segments = self.get_segments_for_chunk(str(chunk_path))
+            all_segments.extend(self.add_offset_to_segments(segments, offset))
+
+        shutil.rmtree(audio_chunk_path)
+        return self.convert_segments(all_segments)
+
+    @staticmethod
+    def convert_segments(segments: List[GroqTranscriptionSegment]) -> List[Segment]:
+        return [
+            Segment(
+                start=seg.start,
+                end=seg.end,
+                text=seg.text,
+            )
+            for seg in segments
+        ]
+
+    @staticmethod
+    def add_offset_to_segments(
+        segments: List[GroqTranscriptionSegment], offset_ms: int
+    ) -> List[GroqTranscriptionSegment]:
+        offset_sec = float(offset_ms) / 1000.0
+        for segment in segments:
+            segment.start += offset_sec
+            segment.end += offset_sec
+
+        return segments
+
+    def get_segments_for_chunk(self, chunk_path: str) -> List[GroqTranscriptionSegment]:
+        retries = 0
+        backoff_time = self.initial_backoff
+
+        while True:
```

Comment:
I noticed that there is a `groq` api client [library](https://pypi.org/project/groq/) which would probably give us retries and things for free. Did you consider using that? Slight preference towards using that unless there's a reason not to.